src/core/thread/mle_router.cpp

```diff
@@ -2134,6 +2156,10 @@ otError MleRouter::UpdateChildAddresses(const Message &aMessage, uint16_t aOffse
         Get<AddressResolver>().Remove(address);
     }
 
+#if OPENTHREAD_CONFIG_TMF_PROXY_MLR_ENABLE
+    Get<MlrManager>().UpdateProxiedSubscriptions(aChild, oldMlrRegisteredAddresses, oldMlrRegisteredAddressNum);
```

Comment:
If I understand it correctly, in the code above, we go through all child addresses and find all (old) registered addresses, copy them into an array and then pass the array here to `UpdateProxiedSubscriptions()` along with the `aChild`.

Can we consider having the `UpdateProxiedSubscriptions()` perform this (since we are passing `aChild` to it already).

Reply:
You understood correctly.

We must save old registered addresses and pass them to  `UpdateProxiedSubscriptions` because they are rewritten in this function. 

If we decide to `UpdateProxiedSubscriptions` before old addresses are cleared in `Child`, then we need to retrieve all new addresses and pass them to  `UpdateProxiedSubscriptions`. I think this is complicated because the new addresses could be compressed so we need to extract them. We also can not rewrite the two old masks (i.e. `mMlrRegisteredMask` and `mMlrToRegisterMask`) because they are still useful.

Reply:
I see. We deleter the addresses. got it.